cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md

```diff
@@ -285,19 +285,25 @@ Unquoted attributes can be broken out of with many characters including `[space]
 
 Also, the `</style>` tag will close the style block even though it is inside a quoted string because the HTML parser runs before the JavaScript parser. Please note that we recommend aggressive CSS encoding and validation to prevent XSS attacks for both quoted and unquoted attributes.
 
-### RULE \#5 - URL Encode Before Inserting Untrusted Data into HTML URL Parameter Values
+### RULE \#5 - URL Encode Before Inserting Untrusted Data into URL Parameter Values
 
-Rule \#5 is for when you want to put untrusted data into HTTP GET parameter value.
+This rule is for when you want to put untrusted data into URL parameter values. It is adviced to follow this rule to prevent from [HTTP Parameter Pollution](https://owasp.org/www-pdf-archive/AppsecEU09_CarettoniDiPaola_v0.8.pdf). HTTP Parameter Pollution itself can be dangerous enough and also it can increase an XSS flaw’s impact in some ways.
```

Comment:
I do not understand why you mentioned here Parameter Pollution? Parameter Pollution is different vulnerability and you mitigate it in different ways but in general you want to fix Parameter Pollution on the backend.
You want to properly endcode URL to prevent XSS, mainly DOM XSS where javascript code takes URL fragment (#) without sanitization.

Reply:
To be honest I didn't want mention HTTP Parameter Pollution in XSS document, but I couldn't find any example of XSS that html/attribute/javascript encoding cannot stop but url encoding can :l and now thinking about fragment but couldn't imagine any example.